[MIG] report_qweb_signer: Migration to 12.0

chienandalu opened the pull request:
This module extends the functionality of report module to sign
PDFs using a PKCS#12 certificate.

cc @Tecnativa

antespi committed:
report_qweb_signer addon

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
report_qweb_signer: Add feature request

pedrobaeza committed:
[MIG] report_qweb_signer: Migration to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

JordiBForgeFlow committed:
fix test_report_qweb_signer - tests were causing travis to stall

yajo committed:
[FIX] report_qweb_signer: Remove hardcoded memory limits

These limits were being hit when printing PDF reports with just 80 pages.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update report_qweb_signer.pot

pedrobaeza committed:
[FIX] README syntax

chienandalu committed:
[MIG] report_qweb_designer: Migration to 12.0

pedrobaeza commented:
Please check tests failure and runbot warning:
```
2018-11-13 09:49:28,314 152 WARNING openerp_test odoo.models: The model report.certificate has no _description
```

chienandalu commented:
I have the tests failing on java's memory: https://travis-ci.org/OCA/reporting-engine/jobs/454946302#L783 Any travis configurations needed?

pedrobaeza committed:
.travis.yml: Adapt to latest standard + add variable for JavaVM error

Avoid error:

Error occurred during initialization of VM
Could not allocate metaspace: 1073741824 bytes

trying solution proposed in https://stackoverflow.com/a/33794801

pedrobaeza commented:
I'm trying one possible patch.

pedrobaeza commented:
No luck. @Yajo do you guess something?

yajo commented on report_qweb_signer/data/defaults.xml:
```diff
@@ -0,0 +1,7 @@
+<odoo>
+    <record model="ir.config_parameter" id="report_qweb_signer_java_param">
+        <field name="key">report_qweb_signer.java_parameters</field>
+        <field name="value">-Xms4M -Xmx4M</field>
```
Try removing this default.

See https://github.com/OCA/reporting-engine/pull/253/commits/c82a549b72edcae277171f2db3518ad4183dfcd8

chienandalu commented:
@Yajo Still the same: https://travis-ci.org/OCA/reporting-engine/jobs/456961880#L783

yajo commented:
You can try adding this to `.travis.yml`, as explained in https://github.com/travis-ci/travis-ci/issues/8408:

```yaml
before_script:
  - _JAVA_OPTIONS=
```

chienandalu commented:
Thanks @Yajo let's try that.

chienandalu commented:
Nay :disappointed:

yajo commented:
Could you try ideas from https://stackoverflow.com/a/33794801/1468388?

You could try running `ulimit -v` or similar inside Travis to know its limits...

pedrobaeza committed:
[MIG] report_qweb_signer: Refine migration

- Good dependency chain
- Context for forcing rendering PDF
- Extra test

pedrobaeza commented:
Merging as it's finally green

etobella commented:
Just a question, I found a python library that could sign documents:
https://github.com/m32/endesive
Maybe we could check if we can replace the java. What do you think?

pedrobaeza commented:
Well, this is working fine since a lot, so although Java is not my favorite one, unless this is proven to be work painless, I won't switch. We can try on the CS.

etobella commented:
Ok, we will check it there :wink:

chienandalu commented:
Hi, @etobella Yes, I discarded to try it because the project seemed to be a little immature yet. There was no much feedback or documentation around so adopting it seemed a little bit risky.
